Consider adding `_CRT_SECURE_NO_WARNINGS` to disable warnings on Windows.

### New Issue Checklist

- [X] I'm using the latest version of Swift Crypto (`main` branch)
- [X] I read the [Contribution Guidelines](https://github.com/apple/swift-crypto/blob/master/CONTRIBUTING.md)
- [X] I searched for [existing GitHub issues](https://github.com/apple/swift-crypto/issues)

### Expected behavior
Warning free build on Windows.

### Actual behavior
Some code uses what Microsoft has marked as "insecure CRT functions".  This produces a lot of warning output on Windows.  Due to SPM's inability to pass `cSettings` to dependencies, I can't disable these warnings.  

Please add `.define("_CRT_SECURE_NO_WARNINGS")` to your Package.swift to disable the warnings.

```
swift-crypto\Sources\CCryptoBoringSSL\crypto\x509v3\v3_utl.c:1173:7: warning: 'sscanf' is deprecated: This function or variable may be unsafe. Consider using sscanf_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [-Wdeprecated-declarations]
  if (sscanf(in, "%d.%d.%d.%d", &a0, &a1, &a2, &a3) != 4) {
      ^
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\stdio.h:2239:20: note: 'sscanf' has been explicitly marked deprecated here
    _Check_return_ _CRT_INSECURE_DEPRECATE(sscanf_s)
                   ^
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\vcruntime.h:355:55: note: expanded from macro '_CRT_INSECURE_DEPRECATE'
        #define _CRT_INSECURE_DEPRECATE(_Replacement) _CRT_DEPRECATE_TEXT(    \
                                                      ^
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\vcruntime.h:345:47: note: expanded from macro '_CRT_DEPRECATE_TEXT'
#define _CRT_DEPRECATE_TEXT(_Text) __declspec(deprecated(_Text))
```

### Steps to reproduce

1. Build on Windows using SPM.

### If possible, minimal yet complete reproducer code (or URL to code)

### Swift Crypto version/commit hash

33a20e650c33f6d72d822d558333f2085effa3dc

### Environment

Stand in the root of the repo and run `./scripts/environment.sh` and paste the output below

<details>
  The script doesn't do anything on Windows, but here's some info:

PS C:\...\swift-crypto> swift --version
compnerd.org Swift version 5.9-dev (LLVM cb96a4a81f9ea98, Swift a57c97bb1a6268e)
Target: x86_64-unknown-windows-msvc

PS C:\...\swift-crypto> systeminfo | Select-String -Pattern "OS Name"
OS Name:                   Microsoft Windows 11 Pro

</details>

--- a/Package.swift
+++ b/Package.swift
@@ -90,10 +90,10 @@ let package = Package(
             ],
             cSettings: [
                 // These defines come from BoringSSL's build system
-                .define("_HAS_EXCEPTIONS", to: "0", .when(platforms: [Platform.windows]),
-                .define("WIN32_LEAN_AND_MEAN", .when(platforms: [Platform.windows]),
-                .define("NOMINMAX", .when(platforms: [Platform.windows]),
-                .define("_CRT_SECURE_NO_WARNINGS", .when(platforms: [Platform.windows]),
+                .define("_HAS_EXCEPTIONS", to: "0", .when(platforms: [Platform.windows])),
+                .define("WIN32_LEAN_AND_MEAN", .when(platforms: [Platform.windows])),
+                .define("NOMINMAX", .when(platforms: [Platform.windows])),
+                .define("_CRT_SECURE_NO_WARNINGS", .when(platforms: [Platform.windows])),
                 /*
                  * These defines are required on Wasm/WASI, to disable use of pthread.
                  */

--- a/Package.swift
+++ b/Package.swift
@@ -89,12 +89,11 @@ let package = Package(
                 "crypto/bio/socket.c"
             ],
             cSettings: [
-                /*
-                 * This define is required on Windows, but because we need older
-                 * versions of SPM, we cannot conditionally define this on Windows
-                 * only.  Unconditionally define it instead.
-                 */
-                .define("WIN32_LEAN_AND_MEAN"),
+                // These defines come from BoringSSL's build system
+                .define("_HAS_EXCEPTIONS", to: "0", .when(platforms: [Platform.windows]),
+                .define("WIN32_LEAN_AND_MEAN", .when(platforms: [Platform.windows]),
+                .define("NOMINMAX", .when(platforms: [Platform.windows]),
+                .define("_CRT_SECURE_NO_WARNINGS", .when(platforms: [Platform.windows]),
                 /*
                  * These defines are required on Wasm/WASI, to disable use of pthread.
                  */

